ext/datadog_profiling_native_extension/private_vm_api_access.c

```diff
@@ -46,6 +46,17 @@
 #else
   #define DDTRACE_UNUSED
 #endif
+extern VALUE eNativeRuntimeError;
+extern VALUE eNativeArgumentError;
+extern VALUE eNativeTypeError;
+// Declare exception class globals from ruby_helpers.h
+// (We can't include ruby_helpers.h here as it pulls in public Ruby headers that conflict with private VM headers)
+#define raise_error(native_exception_class, fmt, ...) \
+  _raise_error(native_exception_class, "" fmt, ##__VA_ARGS__)
+NORETURN(
+  void _raise_error(VALUE native_exception_class, const char *fmt, ...)
+  __attribute__ ((format (printf, 2, 3)));
+);
```

Comment:
Yeah this file is kinda... always a pain. TBH maybe it's easy to just not change anything here -- we'll miss exceptions from this file, but they're rare-ish so I'm not sure we care.

Either that or perhaps having a very simple `extern raise_native_error(char *);` that gets exported elsewhere and that this class uses directly (so this class doesn't get to pick the specific exception class or template stuff -- always get a native error + a static string)"?

Reply:
I'll leave these unchanged for the time being.